### ISSUE TITLE
When in 'interval' mode, the 'weather' forecast screen contains no data

### ISSUE THREAD

**Issue by Graphnic** [GeekMagicClock/smalltv-ultra#36]
When in 'interval' mode, the 'weather forecast' screen contains no data from the feed.  Manually selecting 'weather forecast' manually, the data is loaded. 

Also tried version Ultra-V9.0.19

![IMG_3312](https://github.com/GeekMagicClock/smalltv-ultra/assets/13572989/948e3b94-af29-4cda-a768-65bfa8e917dc)
![IMG_3313 2](https://github.com/GeekMagicClock/smalltv-ultra/assets/13572989/6ed3e03d-1bbc-4bb4-9838-38744c68782e)
![Screenshot 2024-04-10 at 2 10 31 pm](https://github.com/GeekMagicClock/smalltv-ultra/assets/13572989/ce9c3cb7-35c8-49a0-aaf8-56c69c86d73e)

**Comment by GeekMagicClock** [comment]
When the clock is in loop mode, the memory consumption is too large, which may cause some failed requests. You can wait for a while.  In future version, I was considering remove the loop function.

**Comment by Ashwagada** [comment]
> When the clock is in loop mode, the memory consumption is too large, which may cause some failed requests. You can wait for a while. In future version, I was considering remove the loop function.

A step towards better optimization. If the loop is unnecessary, it is worth getting rid of it or changing it to another function.

**Comment by Graphnic** [comment]
> When the clock is in loop mode, the memory consumption is too large, which may cause some failed requests. You can wait for a while.  In future version, I was considering remove the loop function.

You are right, it does sort itself out after a few minutes. I really like the loop feature, so waiting a few minutes isn't an issue! Thanks for the response:-)

**Comment by marcelosuzumura** [comment]
> I really like the loop feature

Me too, it's really useful to cycle automatically between the different views, than just having a fixed one all the time.

**Comment by VaultBoy88** [comment]
It's the same here. After upgrading to .20, the forecast is no longer working. Even rolling back to .19 does not help.

**Comment by Voldenuit23** [comment]
I very much like the loop function as well.

It might be a good idea to wait for all data to be fetched before switching to the forecast screen.

**Comment by GeekMagicClock** [comment]
> I very much like the loop function as well.
> 
> It might be a good idea to wait for all data to be fetched before switching to the forecast screen.

Did you use your own forecast API already?
